tests/tests/amazon_s3/test_s3.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/python
+# Copyright 2016 Mender Software AS
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        https://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+from fabric.api import *
+import pytest
+from common import *
+from common_setup import *
+from helpers import Helpers
+from common_update import common_update_proceduce
+from MenderAPI import adm, deploy
+from mendertesting import MenderTesting
+
+
+@pytest.mark.xfail(reason="Amazon S3 certificate missing")
+@pytest.mark.usefixtures("standard_setup_one_client_bootstrapped_with_s3")
+class TestBasicIntegrationWithS3(MenderTesting):
+
+    @MenderTesting.aws_s3
+    def test_update_image_with_aws_s3(self, install_image=conftest.get_valid_image(), name=None, regnerate_image_id=True):
+        """
+            Perform a successful upgrade using AWS S3, and assert that deployment status/logs are correct.
+
+            A reboot is performed, and running partitions have been swapped.
+            Deployment status will be set as successful for device.
+            Logs will not be retrieved, and result in 404.
+        """
+        if not env.host_string:
+            execute(self.test_update_image_with_aws_s3,
+                    hosts=get_mender_clients(),
+                    install_image=install_image,
+                    name=name,
+                    regnerate_image_id=regnerate_image_id)
+            return
+
+        previous_inactive_part = Helpers.get_passive_partition()
+        deployment_id, expected_image_id = common_update_proceduce(install_image,
+                                                                   name,
+                                                                   regnerate_image_id)
+
+        Helpers.verify_reboot_performed()
+        assert Helpers.get_active_partition() == previous_inactive_part
+        deploy.check_expected_status(deployment_id, "success", len(get_mender_clients()))
+
+        for d in adm.get_devices():
+            deploy.get_logs(d["id"], deployment_id, expected_status=404)
+
+        Helpers.verify_reboot_not_performed()
+        assert Helpers.yocto_id_installed_on_machine() == expected_image_id
```

Comment:
I see what you mean. Final proposal then: Make a separate test, but call this test from within it. That will avoid fixtures for the called function, and will use the fixture from the calling function instead.

As you understand I'm completely allergic to code duplication. :-)